src/arch/Sound/DSoundHelpers.cpp

```diff
@@ -366,58 +362,54 @@ void DSoundBuf::CheckWriteahead( int iCursorStart, int iCursorEnd )
 /* Figure out if we've underrun, and act if appropriate. */
 void DSoundBuf::CheckUnderrun( int iCursorStart, int iCursorEnd )
 {
-	/* If the buffer is full, we can't be underrunning. */
-	if( m_iBufferBytesFilled >= m_iBufferSize )
-		return;
+	// Store the buffer size in a variable to avoid multiple calls
+	int storedSizeOfBuffer = m_iBufferSize;
```

Comment:
This seems unnecessary. It doesn't look like we're changing it and also it's just a plain old data type so seems not required.